supplier_app/views.py

```diff
@@ -273,9 +267,48 @@ def company_join(request, token):
     return redirect('supplier-home')
 
 
-def deny_taxpayer(self, taxpayer_id, request=None):
+def change_taxpayer_status(request, taxpayer_id):
     taxpayer = TaxPayer.objects.get(pk=taxpayer_id)
-    taxpayer.deny_taxpayer()
-    taxpayer.save()
-    taxpayer_notification(taxpayer, 'taxpayer_denial')
+    action = request.POST['action']
+    strategy = get_strategy(action)
+    strategy.change_taxpayer_status(taxpayer)
+    strategy.send_email(taxpayer)
     return redirect('ap-taxpayers')
+
+
+class StrategyStatusChange():
+    def send_email(taxpayer):
+        raise NotImplementedError()
+
+    def change_taxpayer_status(taxpayer):
+        raise NotImplementedError()
+
+
+class StrategyDeny(StrategyStatusChange):
+
+    def send_email(taxpayer):
+        taxpayer_notification(taxpayer, 'taxpayer_denial')
+
+    def change_taxpayer_status(taxpayer):
+        taxpayer.deny_taxpayer()
+        taxpayer.save()
+
+
+class StrategyApprove(StrategyStatusChange):
+
+    def send_email(taxpayer):
+        taxpayer_notification(taxpayer, 'taxpayer_approval')
+
+    def change_taxpayer_status(taxpayer):
+        taxpayer.approve_taxpayer()
+        taxpayer.save()
+
```

Comment:
This could be other file to define each strategy

Reply:
done